### ISSUE TITLE
[no-identical-title] FP when using `test.for`

### ISSUE THREAD

**Issue by jendrikw** [vitest-dev/eslint-plugin-vitest#823]
```
describe("foo", () => {
    test.for([1, 2, 3])("%s", async (val) => {
      // ...
    });
    test.for([4, 5, 6])("%s", async (val) => {
      // ...
    });
});
```

has false positive:

```
 vitest/no-identical-title  Test is used multiple times in the same describe(suite) block
```
